qwandaq/src/main/java/life/genny/qwandaq/utils/BaseEntityUtils.java

```diff
@@ -474,33 +474,67 @@ public BaseEntity addNonLiteralAttributes(BaseEntity entity) {
 		// Handle Created and Updated attributes
 		Attribute createdAttr = new Attribute("PRI_CREATED", "Created", new DataType(LocalDateTime.class));
 		EntityAttribute created = new EntityAttribute(entity, createdAttr, 1.0);
-		created.setValueDateTime(entity.getCreated());
+		// Ensure createdDate is not null
+		try {
+			created.setValueDateTime(entity.getCreated());
+		} catch(NullPointerException e) {
+			log.error("NPE for PRI_CREATED. Generating created date");
+			entity.autocreateCreated();
+			created.setValueDateTime(entity.getCreated());
+		}
 		entity.addAttribute(created);
 
 		Attribute createdDateAttr = new Attribute("PRI_CREATED_DATE", "Created", new DataType(LocalDate.class));
 		EntityAttribute createdDate = new EntityAttribute(entity, createdDateAttr, 1.0);
-		createdDate.setValueDate(entity.getCreated().toLocalDate());
+		// Ensure createdDate is not null
+		try {
+			createdDate.setValueDate(entity.getCreated().toLocalDate());
+		} catch(NullPointerException e) {
+			log.error("NPE for PRI_CREATED_DATE. Generating created date");
+			entity.autocreateCreated();
+			createdDate.setValueDate(entity.getCreated().toLocalDate());
+		}
 		entity.addAttribute(createdDate);
 
+
 		Attribute updatedAttr = new Attribute("PRI_UPDATED", "Updated", new DataType(LocalDateTime.class));
 		EntityAttribute updated = new EntityAttribute(entity, updatedAttr, 1.0);
-		updated.setValueDateTime(entity.getUpdated());
-		entity.addAttribute(updated);
-
-		Attribute updatedDateAttr = new Attribute("PRI_UPDATED_DATE", "Updated", new DataType(LocalDate.class));
-		EntityAttribute updatedDate = new EntityAttribute(entity, updatedDateAttr, 1.0);
-		updatedDate.setValueDate(entity.getUpdated().toLocalDate());
-		entity.addAttribute(updatedDate);
-
-		Attribute codeAttr = new Attribute("PRI_CODE", "Code", new DataType(String.class));
-		EntityAttribute code = new EntityAttribute(entity, codeAttr, 1.0);
-		code.setValueString(entity.getCode());
-		entity.addAttribute(code);
-
-		Attribute nameAttr = new Attribute("PRI_NAME", "Name", new DataType(String.class));
-		EntityAttribute name = new EntityAttribute(entity, nameAttr, 1.0);
-		name.setValueString(entity.getName());
-		entity.addAttribute(name);
+		try {
+			updated.setValueDateTime(entity.getUpdated());
+			entity.addAttribute(updated);
+		} catch(NullPointerException e) {
+			log.error("NPE for PRI_UPDATED");
```

Comment:
How do we figure out what was null? It'd be helpful to print the trace.

Reply:
This is to get us across the line for the demo tomorrow. I have a more robust fix in place on another commit, this is what we agreed on for the time being.

Reply:
Ok. Let's proceed with this PR with the understanding that the handling would be changed in subsequent PR(s).